### DISCUSSION TITLE
Why does this regex for a comment Terminal not work?

### DISCUSSION THREAD

**Discussion by nchammas** [lark-parser/lark#1361] [Q&A] [answered]
Here's a simple demo of what I'm trying to do.

```python
from lark import Lark

parser = Lark(
    r"""
    start: closure
    closure: "(" ")"
    // Works:
    COMMENT: "#" /[^\n]/*
    // Doesn't work:
    // COMMENT: /#.*$/

    %import common.WS
    %ignore WS
    %ignore COMMENT
    """,
    start="start",
)

print(parser.parse(
    """
    ()

    # (
    """
))
```

I'm new to parsing. I started off with what I thought was the obvious way to write a Terminal for comments: `COMMENT: /#.*$/`

This definition is incidentally also used [here in the docs][1].

[1]: https://lark-parser.readthedocs.io/en/stable/recipes.html#collect-all-comments-with-lexer-callbacks

It doesn't work, and the above test yields this error:

```
lark.exceptions.UnexpectedCharacters: No terminal matches '#' in the current parser context, at line 4 col 5

    # (
    ^
```

If I switch the Terminal definition to `COMMENT: "#" /[^\n]/*` (which I picked up from [the docs here][2]), it seems to work fine.

[2]: https://lark-parser.readthedocs.io/en/stable/grammar.html#ignore

What's the difference between these two definitions? And why doesn't the "obvious" regex work?

There is a similar discussion over on #1281.

**Comment by MegaIng** [comment, marked as answer]
`$` matches the end of the string by default. If you want it to match the end of the line, add the `m` flag. `/#.*$/m`. This is clearly documented in the python regex docs.

**Reply by nchammas** [reply]
OK, I see the [documentation for `m` here][1].

So I suppose that, since `.` doesn't match newlines, I can even rewrite the Terminal as `/#.*/`, without the `$`.

(This is also how it's done in [the Lark docs I referenced above][2]--I just missed the lack of a `$` at the end, which drove part of my confusion.)

[1]: https://docs.python.org/3/library/re.html#re.MULTILINE
[2]: https://lark-parser.readthedocs.io/en/stable/recipes.html#collect-all-comments-with-lexer-callbacks
